File: Samples/visio_managed_code_interop_vs2010/VS1010_CSharp_Samples/page_setresults.cs
using System.Collections.Generic;
using VisioInterop;
using IVisio = Microsoft.Office.Interop.Visio;

public static partial class VS2010_CSharp_Samples
{
    public static void Page_SetResults(IVisio.Document doc)
    {
        var page = VisioInterop.Util.CreateStandardPage(doc, "PSR");
        var shape = VisioInterop.Util.CreateStandardShape(page);
        var request = Util.Create_PageSetResults_Request(shape);

        // MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS


        var SID_SRCStream = new short[request.Length * 4];
        var results_objects = new object[request.Length];
        var unitcodes = new object[request.Length];
        for (int i = 0; i < request.Length; i++)
        {
            SID_SRCStream.Set4(i,request[i].ShapeID, request[i].CellSRC.SectionIndex, request[i].CellSRC.RowIndex, request[i].CellSRC.CellIndex);
            results_objects[i] = request[i].Result;
            unitcodes[i] = request[i].UnitCode;
        }

        // EXECUTE THE REQUEST
        short flags = 0;
        int count = page.SetResults(SID_SRCStream, unitcodes, results_objects, flags);

        // DISPLAY THE INFORMATION
        shape.Text = "SetResults";
    }
}

File: Samples/visio_managed_code_interop_vs2010/Visio2010_VB_Samples/Page_SetResults.vb
Imports IVisio = Microsoft.Office.Interop.Visio


Partial Public Class VS2010_VB_Samples

    Shared Sub Page_SetResults (ByVal doc As Microsoft.Office.Interop.Visio.Document)

        Dim page = VisioInterop.Util.CreateStandardPage(doc, "PSR")
        Dim shape = VisioInterop.Util.CreateStandardShape(page)

        ' CREATE REQUEST
        Dim request = {New With { _
 Key .ID = CShort(shape.ID16), _
 Key .Section = CShort(IVisio.VisSectionIndices.visSectionObject), _
 Key .Row = CShort(IVisio.VisRowIndices.visRowXFormOut), _
 Key .Cell = CShort(IVisio.VisCellIndices.visXFormWidth), _
 Key .UnitCode = CShort(IVisio.VisUnitCodes.visNoCast), _
 Key .Result = CDbl(8.0) _
}, New With { _
 Key .ID = CShort(shape.ID16), _
 Key .Section = CShort(IVisio.VisSectionIndices.visSectionObject), _
 Key .Row = CShort(IVisio.VisRowIndices.visRowXFormOut), _
 Key .Cell = CShort(IVisio.VisCellIndices.visXFormHeight), _
 Key .UnitCode = CShort(IVisio.VisUnitCodes.visNoCast), _
 Key .Result = CDbl(1.3) _
}}

        ' MAP THE REQUEST TO THE STRUCTURES VISIO EXPECTS
        Dim SID_SRCStream = New Short(request.Length * 4 - 1) {}
        Dim results_objects = New Object(request.Length - 1) {}
        Dim unitcodes = New Object(request.Length - 1) {}
        For i As Integer = 0 To request.Length - 1
            SID_SRCStream((i * 4) + 0) = request(i).ID
            SID_SRCStream((i * 4) + 1) = request(i).Section
            SID_SRCStream((i * 4) + 2) = request(i).Row
            SID_SRCStream((i * 4) + 3) = request(i).Cell
            results_objects(i) = request(i).Result
            unitcodes(i) = request(i).UnitCode
        Next

        ' EXECUTE THE REQUEST
        Dim flags As Short = 0
        Dim count As Integer = page.SetResults(SID_SRCStream, unitcodes, results_objects, flags)

        ' DISPLAY THE INFORMATION
        shape.Text = "SetResults"

    End Sub

End Class
